### DISCUSSION TITLE
有人集成过zlib吗？ 编译zlib.so没报错，使用的时候报错了。

### DISCUSSION THREAD

**Discussion by lsx1994** [cloudwu/skynet#1593] [Q&A] [answered]
报错：
<pre>
 lua loader error : error loading module 'zlib' from file 
        ./skyframe/engine/luaclib/zlib.so: undefined symbol: inflate
</pre>

makefile：
<pre>
$(LUA_CLIB_PATH)/zlib.so : 3rd/extend/lua_zlib.c | $(LUA_CLIB_PATH)
	$(CC) $(CFLAGS) $(SHARED) $< -o $@
</pre>

https://github.com/brimworks/lua-zlib/blob/master/lua_zlib.c
系统：ubuntu 20.x lts
系统中已经安装了zlib库，可以成功编译zlib.so
是不是makefile需要添加zlib相关的参数？

**Comment by zhangshiqian1214** [comment, marked as answer]
https://github.com/sundream/ggApp.git 他这个仓库集成了很多库, 里面就有zlib的集成

**Reply by lsx1994** [reply]
多谢

**Reply by lsx1994** [reply]
我全局搜索不到zlib相关， 3rd里面也没有看到， 方便说一下zlib相关集成代码在哪吗？

**Reply by zhangshiqian1214** [reply]
我们用的ggApp里面有集成, 没注意到github上的仓库里没有集成, https://github.com/zhangshiqian1214/skynet-server.git 里面点进skynet里面的3rd里面有

**Reply by lsx1994** [reply]
感谢
